src/render.js

```diff
@@ -1,17 +1,24 @@
-let debug = require('debug')('taskcluster-lib-validate');
-let _ = require('lodash');
+const debug = require('debug')('taskcluster-lib-validate');
+const _ = require('lodash');
+const libUrls = require('taskcluster-lib-urls');
 
-/** Render {$const: <key>} into JSON schema */
-function render(schema, constants) {
+/** 
+ * Render {$const: <key>} into JSON schema and update $ref
+ */
+function render(schema, rootUrl, serviceName, constants) {
   // Replace val with constant, if it is an {$const: <key>} schema
   let substitute = (val) => {
     // Primitives and arrays shouldn't event be considered
     if (!(val instanceof Object) || val instanceof Array) {
       return undefined;
     }
 
+    if (rootUrl && val.$ref) {
+      return {$ref: libUrls.schema(rootUrl, serviceName, val.$ref)};
```

Comment:
This will make every $ref absolute, and does so in a bit of an awkward way (for example, if $ref was already absolute, or even relative but starting with `/`, the result will be wrong.

Perhaps `url.resolve` would be good here?

Reply:
Or maybe we should just require that all $ref's be relative, and not rewrite them?  We wouldn't ever fetch an external resource for a $ref anyway (for example `{$ref: 'https://schemas-emporium.com/verbs.json#'}`}, so a non-relative $ref wouldn't work anyway.

Reply:
They are already defined to be only within a service, so I prefer that we go for option 2 and force them to be relative. I'll update the readme with that.

Reply:
The funny thing is that I think we never really documented any of these specifics before afaict